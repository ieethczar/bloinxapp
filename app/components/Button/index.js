import * as React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet } from "react-native";
import { Button } from 'react-native-elements';

const styles = StyleSheet.create({
  button: {
    marginBottom: 15,
    borderRadius: 20,
  },
});

const Btn = ({ title, type, onPress }) => (
  <Button
    type={type}
    buttonStyle={styles.button}
    title={title}
    onPress={onPress}
  />
);

Btn.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};

export default Btn;
