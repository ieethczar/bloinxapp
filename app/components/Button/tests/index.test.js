import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../index';

describe('Button', () => {
  const props = {
    type: '',
    title: 'some-tittle',
    onPress: jest.fn(),
  };

  it('should render correctly', () => {
    const component = renderer.create(
      <Button {...props} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});