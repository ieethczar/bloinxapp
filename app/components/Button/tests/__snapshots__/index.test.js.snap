// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 3,
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 3,
      },
      undefined,
      false,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "busy": false,
        "disabled": false,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "#2089dc",
          "borderRadius": 20,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 15,
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#2089dc",
            "fontSize": 18,
            "paddingVertical": 1,
            "textAlign": "center",
          }
        }
      >
        some-tittle
      </Text>
    </View>
  </View>
</View>
`;
